frontend/src/components/worksheets/WorksheetTerminal.js

```diff
@@ -186,34 +177,25 @@ class WorksheetTerminal extends React.Component {
                         self.doUIAction(action[0], action[1]);
                     });
                 }
+                return data;
             })
-            .fail(function(jqXHR, status, error) {
+            .catch(function(error) {
                 // Some exception occurred outside of the CLI
-                console.error(jqXHR.responseText);
+                const htmlDoc = new DOMParser().parseFromString(error.response.data, 'text/html');
+                const exception = htmlDoc.getElementsByTagName('pre')[0].innerHTML;
+                return { exception };
             });
     }
     completeCommand(command) {
-        var deferred = $.Deferred();
-        $.ajax({
-            type: 'POST',
-            cache: false,
-            url: '/rest/cli/command',
-            contentType: 'application/json; charset=utf-8',
-            dataType: 'json',
-            data: JSON.stringify({
-                worksheet_uuid: this.props.ws.info.uuid,
-                command: command,
-                autocomplete: true,
-            }),
-            success: function(data, status, jqXHR) {
-                deferred.resolve(data.completions);
-            },
-            error: function(jqHXR, status, error) {
-                console.error(jqHXR.responseText);
-                deferred.reject();
-            },
+        return new Promise((resolve, reject) => {
+            apiWrapper
+                .completeCommand(command, this.props.ws.info.uuid)
+                .then((data) => resolve(data.completions))
```

Comment:
We don't need to construct a new Promise here. We can just remove that and change this line to

```suggestion
                .then((data) => data.completions)
```

and it should automatically return a promise.

Reply:
suggestion applied.

Reply:
suggestion applied.